libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -47,8 +49,13 @@ export class BookSearchComponent {
   }
 
   searchBooks() {
-    if (this.searchForm.value.term) {
-      this.store.dispatch(searchBooks({ term: this.searchTerm }));
+    if (this.searchTerm) {
+      const searchTerm$ = from(this.searchTerm);
+      searchTerm$
+        .pipe(debounceTime(500))
+        .subscribe(
```

Comment:
Since you are subscribing here, you will need a way to unsubscribe, to avoid a memory leak!

Reply:
https://brianflove.com/2016-12-11/anguar-2-unsubscribe-observables/

Reply:
Implemented ngOnDestroy to unsubscribe from observable based on lifecycle.